component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2652,4 +2693,33 @@ private void publishPostEmailOTPValidatedEvent(HttpServletRequest request,
             throw new AuthenticationFailedException(errorMsg, e);
         }
     }
+
+    /*
+     * Checks whether the email address update failure property has been set in the context and returns the boolean value of it.
+     *
+     * @param context Authenticaton context
+     * @return Returns whether the email address update has been failed
+     */
+    private boolean isEmailUpdateFailed(AuthenticationContext context) {
+
+        boolean emailUpdateFailed = false;
+
+        if (context.getProperty(EmailOTPAuthenticatorConstants.EMAIL_UPDATE_FAILURE) != null
+                && Boolean.parseBoolean(String.valueOf(context.getProperty(EmailOTPAuthenticatorConstants.EMAIL_UPDATE_FAILURE)))) {
```

Comment:
Boolean.parseBoolean will handle null check.. so we can remove the following part
(context.getProperty(EmailOTPAuthenticatorConstants.EMAIL_UPDATE_FAILURE) != null

Reply:
Fixed this issue.